src/adapters/EOFeedAdapter.sol

```diff
@@ -37,24 +39,31 @@ contract EOFeedAdapter is IEOFeedAdapter, Initializable {
      * @notice Initialize the contract
      * @param feedManager The feed manager address
      * @param feedId Feed id
-     * @param feedDecimals The decimals of the rate
+     * @param inputDecimals The input decimal precision of the rate
+     * @param outputDecimals The output decimal precision of the rate
      * @param feedDescription The description of feed
      * @param feedVersion The version of feed
      */
     function initialize(
         address feedManager,
         uint16 feedId,
-        uint8 feedDecimals,
+        uint8 inputDecimals,
+        uint8 outputDecimals,
         string memory feedDescription,
         uint256 feedVersion
     )
         external
         initializer
     {
         if (feedManager == address(0)) revert InvalidAddress();
+        if (inputDecimals == 0 || inputDecimals > 18) revert InvalidDecimals();
+        if (outputDecimals == 0 || outputDecimals > 18) revert InvalidDecimals();
         _feedManager = IEOFeedManager(feedManager);
         _feedId = feedId;
-        _decimals = feedDecimals;
+        _outputDecimals = outputDecimals;
+        _inputDecimals = inputDecimals;
+        uint256 diff = inputDecimals > outputDecimals ? inputDecimals - outputDecimals : outputDecimals - inputDecimals;
+        _decimalsDivisor = int256(10 ** diff);
```

Comment:
Why `_decimalsDivisor` is not uint256?

Reply:
It reduces a casting in the normalizePrice calculation